CONTRIBUTING.md

```diff
@@ -0,0 +1,167 @@
+- [Introduction](#introduction)
+- [The contribution process](#the-contribution-process)
+  * [Preparing pull requests](#preparing-pull-requests)
+    1. [Checking the coding style](#checking-the-coding-style)
+    1. [Unit testing](#unit-testing)
+    1. [Building the documentation](#building-the-documentation)
+    1. [Automatic code formatting](#automatic-code-formatting)
+    1. [Signing your work](#signing-your-work)
+    1. [Utility functions](#utility-functions)
+    1. [Backwards compatibility](#backwards-compatibility)
+  * [Submitting pull requests](#submitting-pull-requests)
+- [The code reviewing process (for the maintainers)](#the-code-reviewing-process)
+  * [Reviewing pull requests](#reviewing-pull-requests)
+
+## Introduction
+
+
+Welcome to NVFlare! We're excited you're here and want to contribute. This documentation is intended for individuals and institutions interested in contributing to NVFlare. NVFlare is an open-source project and, as such, its success relies on its community of contributors willing to keep improving it. Your contribution will be a valued addition to the code base; we simply ask that you read this page and understand our contribution process, whether you are a seasoned open-source contributor or whether you are a first-time contributor.
```

Comment:
suggest "we simply ask" -> "we ask" for simplicity.

suggest "whether you are a seasoned open-source contributor or whether you are a first-time contributor." -> "whether you are a seasoned open-source contributor or a first-time contributor." to remove duplicate words.